modules/ROOT/pages/dataweave-type-reuse-module-loader.adoc

```diff
@@ -0,0 +1,432 @@
+DataWeave loads your XSD file and translates declarations inside your file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module. Use them to build new types, type check your variables, do pattern matching, or declare new functions that use types. There are no restrictions on how to use these types.
+
+The following example shows how to look for an XSD file in the path “org/weave/myfolder/User.xsd” being the first path the root of the classpath of your application:
+
+----
+import * from xmlschema!org::weave::myfolder::User
+----
+
+== Import Syntax
+
+In order to import the JSON schema types, as shown above, use the following directive:
+
+----
+import _typesToImport_ from jsonschema!_pathToXsdSchemaFile_
+----
+
+The following example shows how to import a type:
+
+----
+import * from jsonschema!example::schema::Person
+----
+
+Where:
+
+* `typeToImport`: You can replace it by `\*` to import all types defined in the schema or to export a single type from the schema, for example, *Root*. You can also import JSON schema types with a different name, for example, *Root as Person*. This way you can reference the type with that name in the script.
+* `pathToXsdSchemaFile`: This represents the path to the schema file, modifying the file separators for `::` and removing the `.xsd` extension from the file name. For example, if the path to the schema is “example/schema/Person.xsd”, “example::schema::Person” must be used.
+
+== Use Your Types in a DataWeave Script
+
+This is how you can use types to import XML schema files inside your DataWeave script.
+
+.XSD Script
+[source,XML]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <xs:element name="Root">
+        <xs:complexType>
+            <xs:choice>
+                <xs:element name="employee" type="employee"/>
+                <xs:element name="member" type="member"/>
+            </xs:choice>
+            <xs:attribute name="id" use="required"/>
+        </xs:complexType>
+    </xs:element>
+    <xs:complexType name="employee">
+        <xs:sequence>
+            <xs:element name="name" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+
+
+    <xs:complexType name="member">
+        <xs:sequence>
+            <xs:element name="id" type="xs:string"/>
+        </xs:sequence>
+    </xs:complexType>
+</xs:schema>
+----
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from xmlschema!org::mule::weave::v2::module::xmlschema::engine::pattern_matching::User
+
+
+{
+     //This would return true as it complies with the xml schema above.
+      Root @(id:"test") :{
+        employee: {
+            "name": "Mariano"
+        }
+      } match {
+             case is Root -> true
+             else -> false
+        }
+}
+----
+
+== From the XML Model to DataWeave
```

Comment:
Can this title be made more clear?

Reply:
I'm not sure we need it since there's only one subsection for it. Could we replace it with `== Built-in XML Schema Types`?